pandas/core/ops/__init__.py

```diff
@@ -1029,6 +1029,15 @@ def wrapper(self, other, axis=None):
             self._get_axis_number(axis)
 
         res_name = get_op_result_name(self, other)
+        other = lib.item_from_zerodim(other)
+
+        # TODO: shouldn't we be applying finalize whenever
+        #  not isinstance(other, ABCSeries)?
+        finalizer = (
```

Comment:
This is related to the renaming issue I commented about below. 
IIUC, the rename is needed because the finalize sets back the original Series name, which might not be desired. 
I think it would be good to make that clearer (eg in the comments)

> shouldn't we be applying finalize whenever not isinstance(other, ABCSeries)?

Yes, that seems correct to me